### ISSUE TITLE
Wrong types for NuxtConfigurationBuild.typescript

### ISSUE THREAD

**Issue by skyrpex** [nuxt/nuxt.js#5406]
### Version

[v2.5.1](https://github.com/nuxt.js/releases/tag/v2.5.1)

### Reproduction link

[https://codesandbox.io/s/91z3jwqyxw](https://codesandbox.io/s/91z3jwqyxw)

### Steps to reproduce

Run the dev or build commands and check the output.

### What is expected ?

The NuxtConfigurationBuild.typescript typings should be defined. 

### What is actually happening?

The NuxtConfigurationBuild.typescript typings are missing, so the nuxt.config.ts can't compile if we declare the configuration object's type.

### Additional comments?

Here's a workaround:

```ts
// import NuxtConfiguration from "@nuxt/config";

// const config: NuxtConfiguration = {
const config = {
  mode: "spa",
  plugins: ["~/plugins/hello"],
  build: {
    // useForkTsChecker: false,
    typescript: {
      typeCheck: false,
    },
  },
};

export default config;
```

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8952">#c8952</a>)</em></sub></div>

**Comment by danielroe** [comment]
This should be fixed by #5326 in v2.5.2.

**Comment by skyrpex** [comment]
Oh thanks! I tried to look for this issue before but not sure what I did 😅
